Crud therapist test cases

HDerwyn opened the pull request:
added test cases for CRUD therapist

added Therapist folder wich contains all test cases for Therapist CRUD

HDerwyn committed:
TherapistController test cases

HDerwyn committed:
Test CRUD Therapist

HDerwyn committed:
adeed enviroment variables for gitHub actions

Davoyandun commented on package.json:
```diff
@@ -28,7 +28,8 @@
     "postgres": "^3.4.4",
     "sequelize": "^6.37.3",
-    "supertest": "^7.0.0"
+    "supertest": "^7.0.0",
```
Solo es consulta ya que desconozco pero para que usaremos esta librería ??

HDerwyn replied:
esa libreria ya estaba, veo que solo se importa en el test de prueba que se hizo inicialmente en el folder Test/integracion creo ese test de prueba inicial lo realizo Matias y importo esa libreria de supertest.

Davoyandun commented on src/application/use-cases/DeleteTherapistUseCase.js:
```diff
@@ -6,9 +6,14 @@ class DeleteTherapistUseCase {
+    }
+
 
     // Retornar el terapeuta creado
```
Los comentarios me hablan de guardar pero las variables de eliminar

HDerwyn replied:
ese comentario ya estaba por Lautaro, por lo que puse entender hay un repositorio que es el que se comunica con la base de datos, no lo hace ahi directamente, habria que preguntarle a Lautaro como lo manejo

Davoyandun replied:
@lautarobunkowski se encargara de  corregir los comentarios

Davoyandun commented on src/application/use-cases/UpdateTherapistUseCase.js:
```diff
@@ -7,10 +7,10 @@ class UpdateTherapistUseCase {
+    if (isEmptyObject(therapistData)) throw new Error("At least one piece of information is required to update");
 
     // Crear la entidad Terapeuta
     const therapist = new Therapist(
```
Que pasa si un campo viene vacío?? Vamos a sustituir un dato existente por un campo vacío ??

HDerwyn replied:
Entiendo, de la manera en que lo habia previsto era que normalmente los campos de update se validan en el front, la idea es que el front valide que los campos tengan informacion antes de que se haga el submit, por lo que el objeto deberia llegar con la informacion del terapeuta y actualizarse, lo que coloque alli era una validacion adicional con su test respectivo, ahora si se tiene previsto que el front permita un campo vacion en la actualizacion del terapeuta entonces se podria cambiar

HDerwyn committed:
Updated the put method

HDerwyn commented:
Updated the put method to work on update the therapist receiving the object with only the data to update, rest will remain the same. updated the test accordingly.
